nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -144,63 +147,51 @@ void ObstacleLayer::onInitialize()
       clearing_buffers_.push_back(observation_buffers_.back());
     }
 
-    ROS_DEBUG(
+    RCLCPP_DEBUG(node_->get_logger(),
         "Created an observation buffer for source %s, topic %s, global frame: %s, "
         "expected update rate: %.2f, observation persistence: %.2f",
         source.c_str(), topic.c_str(),
         global_frame_.c_str(), expected_update_rate, observation_keep_time);
 
+    rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_default;
+    custom_qos_profile.depth = 50;
```

Comment:
Is this generally needed at 50? Should it be parameterized?

Reply:
50 was the depth in the message filter code in the original, but it could also be a parameter per source I suppose.